.github/workflows/codeql.yml

```
@@ -0,0 +1,46 @@
+name: "CodeQL OpenKAT"
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
```

I think this should also run on release branches and for all PRs, also PRs that have release branches or feature branches as target.

```suggestion
  push:
    branches:
      - "main"
      - "release-*"
  pull_request:
```